src/commands/dev/index.js

```diff
@@ -273,6 +273,8 @@ async function serveRedirect(req, res, proxy, match, options) {
     }
   }
 
+  const staticFile = await getStatic(req.url, options.publicFolder)
+  if (staticFile) req.url = staticFile
```

Comment:
This is done to implement https://docs.netlify.com/routing/redirects/rewrites-proxies/#shadowing behavior on request URL's. If there is a static file, we want to rewrite request URL to match that request. Since we already have `match` object which contains `force` and `to`, we don't need to worry about destination at this point.